taigit/src/libraries/Taiga.ts

```diff
@@ -78,4 +78,118 @@ task_history(taskId : number) : Promise<Object> {
     }
 
     return output;
-}
\ No newline at end of file
+}
+
+/**
+ * @summary This call assess a task based on task_assessment_state_trans based on task Id
+ * @param taskId the ID for the task to assess task
+ * @returns  whether the task's state transition is valid,task status transition, date based on task Id
+ * * {
+ *      state_trans_invalid : Boolean       // true, the task state transition is not valid; false, the task is valid.
+ *      date : number,         // Date and time of history entry in milliseconds since epoch
+ *      user : object,         // Taiga User Object
+ *      status_trans:entry.String[],//Status transition array
+ *
+ * }
+ */
+export async function
+task_assessment_state_trans(taskId : number) : Promise<Object> {
+let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741
+let output : Array<Object> = [];
+for(let entry of data) {
+    let new_entry = {
+        state_trans_invalid : false,
+        date : new Date(entry.created_at).getTime(),
+        user : entry.user,
+        status_trans:entry.values_diff.status
+    }
+
+    //Only three status transistion is valid
+    //which is  ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    if((new_entry.status_trans[0] != "New"&&new_entry.status_trans[1] != "In progress")
+        &&(new_entry.status_trans[0] != "In progress"&&new_entry.status_trans[1] != "Ready for test")
+        &&(new_entry.status_trans[0] != "Ready for test"&&new_entry.status_trans[1] != "Closed"))
```

Comment:
this logic is really hard to follow and im not sure it would work. 
How about: 

```js
    let old_status = new_entry.status_trans[0];
    let new_status = new_entry.status_trans[1];
    if(     (old_status == "New"            && new_status != "In progress") ||
            (old_status == "In progress"    && new_status != "Ready for test") ||
            (old_status == "Ready for test" && new_status != "Closed"))
```

Reply:
If everyone think it is good, it is fine by me. I can change this.

Reply:
I didn't test it with your PR, its just a thought you'll have to make sure it works

Reply:
change it the way as you provided.